trino/client.py

```diff
@@ -62,6 +62,28 @@
 _HEADER_EXTRA_CREDENTIAL_KEY_REGEX = re.compile(r'^\S[^\s=]*$')
 
 
+def disable_header_limit():
+    import http.client
+    import email.parser
+    from http.client import HTTPMessage, _MAXLINE
+
+    def parse_headers(fp, _class=HTTPMessage):
+        headers = []
+        while True:
+            line = fp.readline(_MAXLINE + 1)
+            headers.append(line)
+            if line in (b'\r\n', b'\n', b''):
+                break
+        hstring = b''.join(headers).decode('iso-8859-1')
+        return email.parser.Parser(_class=_class).parsestr(hstring)
+
+    http.client.parse_headers = parse_headers
+
+
+# Disable header limit to avoid LineTooLong('got more than 65536 bytes when reading header line')
+disable_header_limit()
```

Comment:
Could this be moved into e.g `ClientSession` or `TrinoRequest` class?

Reply:
It feels a lot less pythonic to me. it works of course and hides this override in `TrinoRequest` instead of the module's namespace.

```
class TrinoRequest: 
    @staticmethod
    def _disable_header_limit():
        import http.client
        import email.parser
        from http.client import HTTPMessage, _MAXLINE

        def parse_headers(fp, _class=HTTPMessage):
            headers = []
            while True:
                line = fp.readline(_MAXLINE + 1)
                headers.append(line)
                if line in (b'\r\n', b'\n', b''):
                    break
            header_string = b''.join(headers).decode('iso-8859-1')
            return email.parser.Parser(_class=_class).parsestr(header_string)

        http.client.parse_headers = parse_headers

    # Disable header limit to avoid LineTooLong('got more than 65536 bytes when reading header line')
    _disable_header_limit.__func__()
```

Reply:
I created a helper class for this override. Let me know WDYT.